lib/constants/cycles.ts

```diff
@@ -1,58 +1,94 @@
 /**
- * Liturgical Day cycle that can be used as metadata for liturgical days.
+ * Represent the two main cycle, that compose the liturgical year in the Roman rite.
  */
 export const ProperCycles = {
+  /**
+   * The "Proper of Time" (or "Temporal") consists of the movable feasts,
+   * most of them keyed to Easter (which falls on a different Sunday every year),
+   * including Ascension, Pentecost, and so on.
+   */
   ProperOfTime: 'PROPER_OF_TIME',
+  /**
+   * The "Proper of Saints" (or "Sanctorale") consists of the fixed feasts,
+   * celebrated on the very same date each year (no matter what the day of the week),
+   * including Christmas and all the saints' days.
+   */
   ProperOfSaints: 'PROPER_OF_SAINTS',
 } as const;
 
 /**
- * Sundays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link ProperCycles} that contains all the possible proper cycles,
+ * that compose the liturgical year in the Roman rite.
  */
-export const SundaysCycles = {
+export const PROPER_CYCLES = Object.values(ProperCycles);
+
+/**
+ * Represent one of the two main cycles (available from {@link ProperCycles}),
+ * that compose the liturgical year in the Roman rite.
+ */
+export type ProperCycle = typeof PROPER_CYCLES[number];
+
+/**
+ * A three-year cycles for the Sunday mass readings (and some solemnities).
+ * The years are designated A, B, or C. Each yearly cycle begins on the first Sunday of Advent.
+ * Year B follows year A, year C follows year B, then back again to A.
+ */
+export const SundayCycles = {
   YearA: 'YEAR_A',
   YearB: 'YEAR_B',
   YearC: 'YEAR_C',
 } as const;
 
 /**
- * Weekdays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link SundayCycles} that contains all the possible
+ * three-year Sunday cycles.
+ */
+export const SUNDAY_CYCLES = Object.values(SundayCycles);
+
+/**
+ * Represent one of the three-year Sunday cycles (available from {@link SundayCycles}).
+ */
+export type SundayCycle = typeof SUNDAY_CYCLES[number];
+
+/**
+ * A two-year cycle for the weekday mass readings (also called Cycle I and Cycle II).
+ * Odd-numbered years are the year 1 (or Cycle I); even-numbered ones are the year 2 (or Cycle II).
```

Comment:
I personally would refrain to call the cycles a _year_, although I know it is the case in informal speech.

```suggestion
 * There are two cycles for the weekday Mass readings (also called Cycle I and Cycle II).
 * Odd-numbered years are the Cycle I, even-numbered ones are the Cycle II.
```

Reply:
Why do you suggest removing "two-year"? I think it provides a good indication that this cycle is running over two years.

Reply:
Because `two year-long cycles != a two-year cycle`, that’s what I tried to explain in [this comment](https://github.com/romcal/romcal/pull/343#discussion_r767194923).

Reply:
Currently, the value of the `.weekdayCycle` key can be `YEAR_1` or `YEAR_2`. This is why I still want to mention the "year 1" or "year 2" in the documentation.

```suggestion
 * A two-year cycle for the weekday Mass readings (also called Cycle I and Cycle II).
 * Odd-numbered years are the Cycle I (year 1); even-numbered ones are the Cycle II (year 2).
```

Reply:
Okay, it LGTM. However, could we not change `YEAR_{1,2}` to `CYCLE_{1,2}` then?

Reply:
We should use `CYCLE_{I, II}` instead... but I still find `YEAR_{1,2}` more generic (and I prefer it).
I also plan to add localized properties, so in English it will be `Cycle {I, II}`.

Reply:
Okay, do as you wish (I don’t mind it). :wink: